examples/src/test/java/jdocs/guide/ShoppingCartEvents.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+// #guideEvents
+package jdocs.guide;
+
+import java.time.Instant;
+
+public class ShoppingCartEvents {
+  public interface Event extends CborSerializable {
+    String getCartId();
+  }
+
+  public interface ItemEvent extends Event {
+    String getItemId();
+  }
+
+  public static final class ItemAdded implements ItemEvent {
+    public final String cartId;
+    public final String itemId;
+    public final int quantity;
+
+    public ItemAdded(String cartId, String itemId, int quantity) {
+      this.cartId = cartId;
+      this.itemId = itemId;
+      this.quantity = quantity;
+    }
+
+    public String getCartId() {
+      return this.cartId;
+    }
+
+    public String getItemId() {
+      return this.itemId;
+    }
+  }
+
+  public static final class ItemRemoved implements ItemEvent {
+    public final String cartId;
+    public final String itemId;
+    public final int oldQuantity;
+
+    public ItemRemoved(String cartId, String itemId, int oldQuantity) {
+      this.cartId = cartId;
+      this.itemId = itemId;
+      this.oldQuantity = oldQuantity;
+    }
+
+    public String getCartId() {
+      return this.cartId;
+    }
+
+    public String getItemId() {
+      return this.itemId;
+    }
+  }
+
+  public static final class ItemQuantityAdjusted implements ItemEvent {
+    public final String cartId;
+    public final String itemId;
+    public final int newQuantity;
+    public final int oldQuantity;
+
+    public ItemQuantityAdjusted(String cartId, String itemId, int newQuantity, int oldQuantity) {
+      this.cartId = cartId;
+      this.itemId = itemId;
+      this.newQuantity = newQuantity;
+      this.oldQuantity = oldQuantity;
+    }
+
+    public String getCartId() {
+      return this.cartId;
+    }
+
+    public String getItemId() {
+      return this.itemId;
+    }
+  }
+
+  public static final class CheckedOut implements Event {
+    public final String cartId;
+    public final Instant eventTime;
+
+    public CheckedOut(String cartId, Instant eventTime) {
+      this.cartId = cartId;
+      this.eventTime = eventTime;
+    }
+
+    public String getCartId() {
+      return this.cartId;
+    }
+  }
+}
```

Comment:
@seglo which events couldn't be serialized? Two typical gotchas:
* constructor with single parameter: https://doc.akka.io/docs/akka/current/serialization-jackson.html#constructor-with-single-parameter
* `-parameters` Java compiler option https://github.com/akka/akka-samples/blob/2.6/akka-sample-cqrs-java/pom.xml#L136 or for sbt `.settings(javacOptions += "-parameters") // for Jackson`

Reply:
It was a deserializable problem with the Java `ShoppingCartApp`.  The `javacOptions += "-parameters"` setting resolved it.  Thank you!